inference-engine/src/cldnn_engine/cldnn_program.cpp

```diff
@@ -4367,7 +4367,22 @@ void Program::CreateReducePrimitive(cldnn::topology& topology, InferenceEngine::
             static_cast<int32_t>(reduce->keep_dims));
 
     topology.add(reducePrim);
-    AddPrimitiveToProfiler(reduceLayerName, layer);
+
+    auto reorderLayerName = reduceLayerName + "_reorder";
+    cldnn::format out_format = cldnn::format::any;
+    auto out_dt = DataTypeFromPrecision(reduce->outData[0]->getTensorDesc().getPrecision());
+    if (!reduce->keep_dims && reduceDimNumber > 4) {
+        if (reduceDimNumber - rawAxes.size() == 5)
```

Comment:
1. Not realy, reduce->outData[0]->getTensorDesc().getDims().size() doesn't take into account number of reduced axes.
2. I've added 6d case handling, however it will make sense only if in future we'll have 7d or larger formats.